packages/channel-provider/src/messaging-service.ts

```diff
@@ -49,14 +49,20 @@ export class MessagingService {
     }
   }
 
+  private requestNumber = 0;
   async request<ResultType = any>(
     target: Window,
     message: JsonRpcRequest,
     callback?: (result: ResultType) => void
   ): Promise<ResultType> {
-    if (!message.id) {
-      message.id = Date.now();
-    }
+    // Some tests rely on being able to supply the id on the message
+    // We should not allow this in production, as we cannot guarantee unique
+    // message ids.
+    if (message.id) console.error('message id should not be defined');
+
+    // TODO: I don't know the requirements on message IDs, but it's important
+    // that they be unique
```

Comment:
From what I can tell, from the JSON-RPC spec side, this would meet requirements 

> id
An identifier established by the Client that MUST contain a String, Number, or NULL value if included. If it is not included it is assumed to be a notification. The value SHOULD normally not be Null [1] and Numbers SHOULD NOT contain fractional parts [2]

```suggestion
    // message IDs should be unique
```